### DISCUSSION TITLE
[RFC] Pure enums or global enums?

### DISCUSSION THREAD

**Discussion by lmariscal** [cavariux/nimgl#24] [Ideas]
Even though I am not a big fan of the lack of namespace features in Nim and it is clear Araq's position on this take, I have never really ran into any mind boggling issues related to this.
I want to know your take on this topic, the new generator for ImGui binds all the current Enums with the `pure` pragma. Resulting in this type of code:
```nim
io.keyMap[ImGuiKey.LeftArrow.int32] = keyLeft.ord
io.keyMap[ImGuiKey.RightArrow.int32] = keyRight.ord
io.keyMap[ImGuiKey.UpArrow.int32] = keyUp.ord
io.keyMap[ImGuiKey.DownArrow.int32] = keyDown.ord
```
Is this okay with you or would you prefer a more of a C type solution with a 2-3 letter prefix before the enum? With a code that would look like this:
```nim
io.keyMap[igkLeftArrow.int32] = keyLeft.ord
io.keyMap[igkRightArrow.int32] = keyRight.ord
io.keyMap[igkUpArrow.int32] = keyUp.ord
io.keyMap[igkDownArrow.int32] = keyDown.ord
```
(Would have to properly think the prefixes)

My own taste says that the first example is more readable and less error prone. But would like to know your opinion on what you prefer since this change will be applied to all the future NimGL 1.0 generators and bindings.

**Comment by kraptor** [comment]
I really prefer the former one, as global enums sometimes end in name clashes. You know... don't pollute the global namespace :)
